### PR TITLE
NXP S32ZE update to HAL drivers version 1.0.0

### PR DESCRIPTION
**Pull request by congnguyenhuu**
This PR update drivers so that S32ZE works on the Real-Time Drivers 1.0.0

### CODE REVIEW

File: drivers/spi/spi_nxp_s32.c [hunk 1]
```diff
@@ -671,7 +671,7 @@ static const struct spi_driver_api spi_nxp_s32_driver_api = {
 			SPI_MCR_PCSIS(BIT_MASK(SPI_NXP_S32_NUM_CS(n))) |		\
 			SPI_MCR_MDIS(0U) | SPI_MCR_XSPI(1U) | SPI_MCR_HALT(1U)),	\
 		.TransferMode = SPI_IP_POLLING,						\
-		.StateIndex   = SPI_NXP_S32_HW_INSTANCE(n),				\
+		.StateIndex   = n,							\
```
Comment (manuargue) [inline drivers/spi/spi_nxp_s32.c:674]: okay, but this is not related to the HAL version migration. It's an improvement done in this commit https://github.com/zephyrproject-rtos/zephyr/pull/65044/commits/e03a44976861cf1115d6f205e9dbadec8b3d2efe unrelated to the HAL version. So either move it to a separate pr, or update the commit message accordingly to not be misleading.

File: soc/arm/nxp_s32/common/pinctrl_soc.h [hunk 2]
```diff
@@ -19,6 +19,11 @@
 /** @brief Type for NXP S32 pin configuration. */
 typedef Siul2_Port_Ip_PinSettingsConfig pinctrl_soc_pin_t;
 
+#if defined(CONFIG_SOC_SERIES_S32ZE_R52)
```
Comment (manuargue) [inline soc/arm/nxp_s32/common/pinctrl_soc.h:22]: can we do instead?

```c
#if !defined(FEATURE_SIUL2_MAX_NUMBER_OF_INPUT) && defined(FEATURE_SIUL2_MAX_NUMBER_OF_INPUT_U8)
#define FEATURE_SIUL2_MAX_NUMBER_OF_INPUT FEATURE_SIUL2_MAX_NUMBER_OF_INPUT_U8
#endif
```

**Review by manuargue — CHANGES_REQUESTED** [review]

File: soc/arm/nxp_s32/common/pinctrl_soc.h (hunk 2 above)
Comment (congnguyenhuu) [reply soc/arm/nxp_s32/common/pinctrl_soc.h]: Nice. I updated

File: drivers/spi/spi_nxp_s32.c (hunk 1 above)
Comment (congnguyenhuu) [reply drivers/spi/spi_nxp_s32.c:674]: yes, I update commit msg

**Review by manuargue — DISMISSED** [review]

**Review by decsny — DISMISSED** [review]

**Review by Dat-NguyenDuy — DISMISSED** [review]

**Comment by mmahadevan108** [comment]
Please update `west.yml`

**Comment by congnguyenhuu** [comment]
> Please update `west.yml`

I updated west.yml

**Review by manuargue — APPROVED** [review]

**Review by Dat-NguyenDuy — APPROVED** [review]
